src/api/helpers/utils.js

```diff
@@ -44,55 +36,59 @@ export function allowOnlyGetMethod(req, res, next) {
 }
-
-
-export async function getBrowsedRelationships(orm, locals, browsedEntityType,
-											  getEntityInfoMethod, fetchRelated, filterRelationshipMethod) {
+export async function getBrowsedRelationships(
+	orm,
+	locals,
+	browsedEntityType,
+	getEntityInfoMethod,
+	fetchRelated,
+	filterRelationshipMethod
+) {
 	const {entity, relationships} = locals;
-
-	if (!relationships.length > 0) {
+	if (!relationships?.length) {
 		return [];
 	}
-	const relationshipsPromises = relationships
-		.map(async relationship => {
-			let relEntity;
+	try {
+		const relationshipPromises = relationships.map(async (relationship) => {
+			let relEntity = null;
 			if (entity.bbid === relationship.sourceBbid &&
 				relationship.target.type.toLowerCase() === browsedEntityType.toLowerCase()) {
 				relEntity = relationship.target;
 			}
 			else if (relationship.source.type.toLowerCase() === browsedEntityType.toLowerCase()) {
 				relEntity = relationship.source;
 			}
-
-			if (relEntity) {
-				const loadedRelEntity = await loadEntity(orm, relEntity, fetchRelated);
-				const formattedRelEntity = getEntityInfoMethod(loadedRelEntity);
-				if (!filterRelationshipMethod(formattedRelEntity)) {
-					return null;
-				}
-				return {
-					entity: formattedRelEntity,
-					relationship: [{
-						relationshipType: _get(relationship, 'type.label', null),
-						relationshipTypeID: _get(relationship, 'type.id', null)
-					}]
-				};
+			if (!relEntity) {
+				return null;
+			}
+			const loadedRelEntity = await loadEntity(orm, relEntity, fetchRelated);
+			const formattedRelEntity = getEntityInfoMethod(loadedRelEntity);
+			if (!filterRelationshipMethod(formattedRelEntity)) {
+				return null;
 			}
-			return null;
+			return {
+				entity: formattedRelEntity,
+				relationships: [{
+					relationshipType: _get(relationship, 'type.label', null),
+					relationshipTypeID: _get(relationship, 'type.id', null)
+				}]
+			};
 		});
-	const fetchedRelationshipsPromises = await Promise.all(relationshipsPromises);
-	// Remove falsy values (nulls returned above)
-	const filteredRelationships = fetchedRelationshipsPromises.filter(Boolean);
-
-	return filteredRelationships
-		.reduce((accumulator, relationship) => {
-			const entityAlreadyExists = accumulator.find(rel => rel.entity.bbid === relationship.entity.bbid);
-			if (entityAlreadyExists) {
-				entityAlreadyExists.relationships.push(...relationship.relationships);
+		const results = await Promise.all(relationshipPromises);
+		const filteredResults = results.filter(Boolean);
+		return filteredResults.reduce((accumulator, relationship) => {
+			const existingEntity = accumulator.find(rel => rel.entity.bbid === relationship.entity.bbid);
+			if (existingEntity) {
+				existingEntity.relationships.push(...relationship.relationships);
 			}
 			else {
 				accumulator.push(relationship);
 			}
 			return accumulator;
 		}, []);
+	}
+	catch (error) {
+		const errorMessage = 'Failed to fetch browsed relationships';
```

Comment:
It could be helpful to have the full error message here for debugging purposes:
```suggestion
		const errorMessage = `Failed to fetch browsed relationships: ${error.toString()}`;
```